executor/aggregate.go

```diff
@@ -671,7 +673,7 @@ func (w *HashAggFinalWorker) consumeIntermData(sctx sessionctx.Context) (err err
 			memSize := getGroupKeyMemUsage(w.groupKeys)
 			w.groupKeys = w.groupKeys[:0]
 			for i := 0; i < groupKeysLen; i++ {
-				w.groupKeys = append(w.groupKeys, []byte(groupKeys[i]))
+				w.groupKeys = append(w.groupKeys, hack.Slice(groupKeys[i]))
 			}
 			failpoint.Inject("ConsumeRandomPanic", nil)
 			w.memTracker.Consume(getGroupKeyMemUsage(w.groupKeys) - memSize)
```

Comment:
Only need consider the slice size here. We don't need to consume the memory usage of string's data, because L676 is shallow copy.